File: helpdeskspringboot/src/main/java/com/ez/service/PriorityService.java
package com.ez.service;

import com.ez.entity.Priority;
import com.ez.repository.PriorityRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PriorityService {

    private Logger LOGGER = LoggerFactory.getLogger(getClass());

    @Autowired
    private PriorityRepository priorityRepository;

    // search priorities by pageNumber and based on the search criteria.
    // parameters:
    //  - pageNumber: page number
    //  - pageSize: page size
    //  - searchTerm: ID, name
    //  - reachInOpt: >=(gt), =(eq), <=(lt)
    //  - reachIn: number of hours to complete a ticket
    //  - status: '', 'Active', 'Inactive'
    public List<Priority> searchPriorities(int pageNumber, int pageSize, String searchTerm, String reachInOpt, long reachIn, String status) {

        LOGGER.info("search categories");

        return priorityRepository.searchPriorities(pageNumber, pageSize, searchTerm, reachInOpt, reachIn, status);
    }

    // calculate total of priorities based on the search criteria
    public long getTotalOfPriorities(String searchTerm, String reachInOpt, long reachIn, String status) {
        LOGGER.info("get total of priorities");

        return priorityRepository.getTotalOfPriorities(searchTerm, reachInOpt, reachIn, status);
    }

//    // create new category
//    public Category createCategory(Category category) {
//
//        LOGGER.info("create new category");
//
//        // new user
//        Category newCategory = new Category(category.getName(), category.getStatus());
//
//        // save new category into database
//        categoryRepository.save(newCategory);
//
//        return newCategory;
//    }
//
//    // find category by category id
//    public Category findById(Long id) throws CategoryNotFoundException {
//
//        LOGGER.info("find category by id");
//
//        // find category by category id
//        return categoryRepository.findById(id).orElseThrow(() -> new CategoryNotFoundException(NO_CATEGORY_FOUND_BY_ID + id));
//    }
//
//    // update existing category
//    public Category updateCategory(Category category) throws CategoryNotFoundException {
//
//        LOGGER.info("Update category");
//
//        // get existing category(persistent)
//        Category existingCategory = categoryRepository.findById(category.getId())
//                .orElseThrow(() -> new CategoryNotFoundException(NO_CATEGORY_FOUND_BY_ID + category.getId()));
//
//        // set new values to existing category
//        existingCategory.setName(category.getName());
//        existingCategory.setStatus(category.getStatus());
//
//        // update existing category(persistent)
//        categoryRepository.save(existingCategory);
//
//        return existingCategory;
//    }

}

File: helpdeskangular/src/app/service/priority.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Priority } from '../entity/Priority';

@Injectable({
  providedIn: 'root'
})
export class PriorityService {

  // 'http://localhost:8080'
  host = environment.apiUrl;

  // number of priorities per a page(default = 5)
  pageSize = environment.pageSize;

  constructor(private http: HttpClient) { }

  // get priorities by page and based on the search criteria
  // parameters:
  //  - pageNumber: page number
  //  - searchTerm: search term(ID, name)
  //  - reachInOpt: gt(>=), eq(=), lt(<=)
  //  - reachIn: number of hours to complete a ticket
  //  - status: '', 'Active', 'Inactive'  
  searchPriorities(pageNumber: number, searchTerm: string, reachInOpt: string, reachIn: number, status: string): Observable<Priority[]> {

    console.log(`${this.host}/priority-search?pageNumber=${pageNumber}&pageSize=${this.pageSize}&searchTerm=${searchTerm}&reachInOpt=${reachInOpt}&reachIn=${reachIn}&status=${status}`);

    // ex: url = http://localhost:8080/priority-search?pageNumber=0&pageSize=5&searchTerm=""&reachInOpt="gt"&reachIn=0&status=""
    return this.http.get<Priority[]>(
      `${this.host}/priority-search?pageNumber=${pageNumber}&pageSize=${this.pageSize}&searchTerm=${searchTerm}&reachInOpt=${reachInOpt}&reachIn=${reachIn}&status=${status}`
    )
  }

  // calculate total of priorities for count total pages
  // parameters:
  //  - searchTerm: search term(ID, name)
  //  - reachInOpt: gt(>=), eq(=), lt(<=)
  //  - reachIn: number of hours to complete a ticket
  //  - status: '', 'Active', 'Inactive'
  getTotalOfPriorities(searchTerm: string, reachInOpt: string, reachIn: number, status: string): Observable<number> {

    // ex: http://localhost:8080/total-of-priorities?searchTerm=""&reachInOpt="gt"&reachIn=0&status=""
    return this.http.get<number>(
      `${this.host}/total-of-priorities?searchTerm=${searchTerm}&reachInOpt=${reachInOpt}&reachIn=${reachIn}&status=${status}`
    );

  }

  // create priority
  public createPriority(priority: Priority): Observable<Priority> {
    return this.http.post<Priority>(`${this.host}/priority-create`, priority);
  }

  // // edit existing category
  // public editCategory(category: Category): Observable<Category> {
  //   return this.http.put<Category>(`${this.host}/category-edit`, category);
  // }

  //  // find category by id
  //  findById(id: number): Observable<Category> {
  //    return this.http.get<Category>(`${this.host}/category-list/${id}`);
  //  }

}